src/Checkout/Payment/Service/VaultTokenService.php

```diff
@@ -132,18 +155,41 @@ public function requestVaulting(VaultablePaymentSourceInterface $paymentSource):
         $paymentSource->setAttributes($attributes);
     }
 
-    private function saveTokenToSubscription(SubscriptionEntity $subscription, string $tokenId, string $paymentMethodId, Context $context): void
+    /**
+     * @internal
+     */
+    public function isSubscriptionContext(
+        ?SalesChannelContext $context = null,
+        ?ParameterBag $bag = null,
+        ?PaymentTransactionStruct $paymentTransaction = null,
+    ): bool {
+        if ($context && ($context->hasExtension('subscription') || (\class_exists(PlanIntervalMappingStruct::class) && $context->getExtensionOfType('subscriptionManagedContexts', PlanIntervalMappingStruct::class)?->all()))) {
+            return true;
+        }
+
+        if ($bag && $bag->getBoolean(VaultTokenService::REQUEST_CREATE_VAULT)) {
```

Comment:
I don't like this. The request parameter has nothing to do with subscriptions. Maybe call the method `shouldRequestVaulting`?